indico/modules/events/registration/fields/simple.py

```diff
@@ -472,7 +472,7 @@ def _stay_dates_valid(form, field):
                 raise ValidationError(_("Arrival date can't be set after the departure date."))
 
         def _check_number_of_places(form, field):
-            if not field.data:
+            if not field.data or not field.data.get('choice'):
```

Comment:
I think I can change it without any harm since there are no invalid records in the DB.

```
select rd.*
from event_registration.registration_data rd
join event_registration.form_field_data ffd on ffd.id = rd.field_data_id
join event_registration.form_items fi on ffd.field_id = fi.id
where fi.input_type = 'accommodation'
and not fi.is_deleted
and (rd.data->>'choice') is NULL;
```